### PR TITLE
Add functionality to perform regression

### LINKED ISSUE #154: Support multi-class & continuous outcomes
This is probably a bit lower on the priority list.

If more than two outcome classes, use caret multiClass and check that logit models isn't being used.

Edit: can we support continuous outcomes also?

- [x] Add multinomial logistic regression (for multiclass outcomes).
- [x] Add linear regression (for continuous outcomes).
- [x] Check to see if decision tree, random forest, and xgboost code works for multiclass and continuous; modify if needed (including check defaults to see if they're good ones to use).
- [x] Have default performance metric for each class (AUROC for binary, RMSE for continuous, logLoss for multiclass?).
~- [ ] Be clear in the vignette what models can be used for what types of outcome.~ (because all of them can be used for everything now; except maybe SVM...)
- [x] Change checks to work for non-binary data.
    - [x] Have warning for numeric data that looks like it might be binary or multiclass (related to #166).
- [x] Use `perf_metric_fn` to calculate test performance metrics.

### LINKED ISSUE #166: Throw more informative error if outcome values are numbers
Example error for numeric 0,1 outcome:
```
Using L2 normalization for Logistic Regression
Error: wrong model type for regression
```

Example error for factor 0,1 outcome:
```
Using L2 normalization for Logistic Regression
Error: At least one of the class levels is not a valid R variable name; This will cause errors when class probabilities are generated because the variables names will be converted to  X0, X1 . Please use factor levels that can be used as valid R variable names  (see ?make.names for help).
```

### PR DESCRIPTION
**Pull request by zenalapp**
Resolves #154 and resolves #166 

- Changed `regLogistic` to `glmnet` because you can perform both classification and regression using `glmnet`.
- Added checks for numeric outcome.
- Added ntree argument so that users can change it for random forest if desired.
- Updated tests.

### CODE REVIEW

File: R/train_model.R [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+#' @param train_data train_data
+#' @param method method
+#' @param cv cross-validation caret scheme
+#' @param perf_metric_name performance metric name
+#' @param tune_grid tuning grid
+#' @param ntree number of trees for random forest
+#'
+#' @return trained model
+#' @export
+#'
```
Comment (kelly-sovacool) [inline R/train_model.R:15]: Can you use the `@author` roxygen directive here

File: R/train_model.R [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+#' Train model
+#'
+#' @param model_formula model formula
+#' @param train_data train_data
+#' @param method method
+#' @param cv cross-validation caret scheme
+#' @param perf_metric_name performance metric name
+#' @param tune_grid tuning grid
+#' @param ntree number of trees for random forest
```
Comment (kelly-sovacool) [inline R/train_model.R:11]: recommend using `@inheritParams` to avoid copy-pasting documentation (.eg. `run_ml()` already has some of these)

File: data-raw/otu_mini.R [hunk 3]
```diff
@@ -5,56 +5,41 @@ usethis::use_data(otu_mini, overwrite = TRUE)
...
-  "I", "D", "A", "C", "D", "H", "A", "A", "J", "F", "E", "C", "J",
-  "J", "G", "C", "D", "H", "E", "E", "F", "G", "F", "C", "E", "F",
-  "D", "D", "B", "J", "B", "H", "A", "A", "A", "B", "D", "J", "D",
-  "F", "F", "B", "G", "J", "B", "F", "G", "F", "J", "B", "D", "B",
-  "C", "C", "H", "B", "F", "I", "G", "I", "D", "G", "G", "E", "F",
-  "I", "B", "B", "I", "J", "A"
-)
-otu_mini_results1 <- mikropml::run_ml(otu_mini, # use built-in hyperparams
-  "regLogistic",
+otu_mini_group <- sample(LETTERS[1:5], nrow(otu_small), replace = TRUE)
```
Comment (kelly-sovacool) [inline data-raw/otu_mini.R:35]: I was getting inconsistent results with `sample()` even with `set.seed()` immediately before it. So I pasted its `dput()` results instead of having data-raw run it again every time.

File: data-raw/otu_mini_continuous.R [hunk 4]
```diff
@@ -0,0 +1,85 @@
+## code to prepare `otu_mini` dataset
+set.seed(2019)
+outcome_colname <- "Otu00001"
+kfolds <- 2
+otu_mini <- otu_small[, 1:4]
+# usethis::use_data(otu_mini, overwrite = TRUE)
+
+test_hyperparams <- structure(list(
```
Comment (kelly-sovacool) [inline data-raw/otu_mini_continuous.R:8]: Is there a reason you're defining `test_hyperparams` here instead of letting `run_ml()` use the hyperparams function?

File: data-raw/otu_small.R [hunk 5]
```diff
@@ -8,8 +8,7 @@ usethis::use_data(otu_small, overwrite = TRUE)
 ## code to prepare models with the `otu_small` otu_small
 set.seed(2019)
 outcome_colname <- "dx"
-outcome_value <- "cancer"
-kfolds <- 5
+kfolds <- 2
```
Comment (kelly-sovacool) [inline data-raw/otu_small.R:11]: Probably should just delete these variables since they don't actually get used.

**Review by kelly-sovacool — APPROVED** [review]

**Review by kelly-sovacool — CHANGES_REQUESTED** [review]
Oops I meant to select "Request changes" instead of "Approve"

File: data-raw/otu_mini_continuous.R (hunk 4 above)
Comment (kelly-sovacool) [reply data-raw/otu_mini_continuous.R:8]: We should probably just get rid of `test_hyperparams`, they're leftover from the time before we had functions to get default hyperparameters dynamically.

**Comment by kelly-sovacool** [comment]
Re: the failed pr_build: can you try running `devtools::build_site()` locally and see if it works? There may be a problem with one of the vignettes.

File: R/train_model.R (hunk 1 above)
Comment (zenalapp) [reply R/train_model.R:16]: Done

File: R/train_model.R (hunk 2 above)
Comment (zenalapp) [reply R/train_model.R]: Done

File: data-raw/otu_mini.R (hunk 3 above)
Comment (zenalapp) [reply data-raw/otu_mini.R]: Done. Letters are different, but now hard-coded.

File: data-raw/otu_small.R (hunk 5 above)
Comment (zenalapp) [reply data-raw/otu_small.R:11]: Deleted kfolds, but we do use outcome_colname so I kept that one. We can change more if we want when we clean up how we're organizing the data.

**Comment by zenalapp** [comment]
`devtools::build_site()` worked locally @kelly-sovacool. Also, addressed all of your comments.

File: data-raw/otu_mini_continuous.R (hunk 4 above)
Comment (zenalapp) [reply data-raw/otu_mini_continuous.R:8]: I think 1) it takes longer with all the hyperparameters, and 2) I was getting warnings, but now that we're going to fix that it's probably okay to use all of them. Think we should fix that here or in a separate branch?

**Comment by kelly-sovacool** [comment]
@zenalapp did you decide not to get rid of the hard-coded test hyperparameters?

**Comment by kelly-sovacool** [comment]
Also @zenalapp can you commit the `docs/` after running `devtools::build_site()` so the website will get updated?

**Comment by zenalapp** [comment]
- I accidentally responded to this late: I think 1) it takes longer with all the hyperparameters, and 2) I was getting warnings, but now that we're going to fix that it's probably okay to use all of them. Think we should fix that here or in a separate branch?
- Pushed all the changes.
@kelly-sovacool

**Review by kelly-sovacool — APPROVED** [review]
Let's merge this as-is, and if/when we address the uninformative error messages later we can nix the hard-coded hyperparameters if we want to.
